### PR TITLE
add CloudFunction example to sync Gsuite w/ stackdriver

### PR DESCRIPTION
**Pull request by cleibl**
Created a Terraform Module to use the Gsuite Exporter Python Library in a CloudFunction.

The Cloud Scheduler and CloudFunction Terraform Resources are not fully supported (or do not have all arguments supported) so we had to use gcloud commands for those resources.

I've added this to the examples folder, but I think it might need to be its own repository

### CODE REVIEW

**Comment by morgante** [comment]
@ocervell  Can you take a look at this?

@cleibl It looks like the Python code was pulled from [here](https://github.com/GoogleCloudPlatform/professional-services/tree/master/tools/gsuite-exporter). Is there a way we could approach this without forking the codebase?

**Comment by cleibl** [comment]
We could run a "local-exec" and pull latest from the repository during initial run.  It would be best if we could pull **just** the python library and not the entire PSO repository, but not critical  

I'd have to look into if it is possible to pull specific directories in a repository with pulling the entire repository.  I think this capability is available.

**Comment by morgante** [comment]
@cleibl How about if we publish the library to PyPi?

**Comment by cleibl** [comment]
@morgante Yes that would also work! I believe we could just add the library name into 'requirements.txt' after the library available in PyPi.  Probably the best path forward

**Comment by ocervell** [comment]
We'll work to publish the `gsuite-exporter` to PyPI. It's probably a better approach than copying its code in the `examples/` folder right now.

**Review by aaron-lane — CHANGES_REQUESTED** [review]
@ocervell do you have any update on publishing the Python module?

**Comment by ocervell** [comment]
@cleibl @aaron-lane sorry for the delayed response here: yes, the package is now published to PyPI (https://pypi.org/project/gsuite-exporter/) and can now be installed using `pip install gsuite-exporter`. 

I will shortly make a PR in this module to address the source of the package in the main module.

@cleibl feel free to clean up unwanted code (duplication of the package) and update this PR.

**Comment by cleibl** [comment]
@ocervell @aaron-lane Updated the CFN to remove local code and point to package in PyPI.   Additionally, removed the workaround local execs now that CFN and Cloud Scheduler Resources are supported and upgraded to 0.12

File: examples/CloudFunction/README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    - `admin` - [Admin activity reports](https://developers.google.com/admin-sdk/reports/v1/guides/manage-audit-admin.html)
+    - `drive` - [Google Drive activity reports](https://developers.google.com/admin-sdk/reports/v1/guides/manage-audit-drive.html)
+    - `login` - [Login activity reports](https://developers.google.com/admin-sdk/reports/v1/guides/manage-audit-login.html)
+    - `mobile` - [Mobile activity reports](https://developers.google.com/admin-sdk/reports/v1/guides/manage-audit-mobile.html)
+    - `token` - [Authorization Token activity reports](https://developers.google.com/admin-sdk/reports/v1/guides/manage-audit-tokens.html)
+
+
+The following destinations are currently supported:
+- [Stackdriver Logging](https://cloud.google.com/logging/docs/)
+
```
Comment (ocervell) [inline examples/CloudFunction/README.md]: Lines 5 - 29: Please don't copy paste the gsuite-exporter documentation but link to the [gsuite-exporter docs](https://github.com/GoogleCloudPlatform/professional-services/tree/master/tools/gsuite-exporter) instead.

File: examples/CloudFunction/gsuite-exporter-cloudfunction/main.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import json
+from gsuite_exporter.cli import sync_all
+
+def run (data, context):
+
+    # Decode Data From Pubsub Message
+    name = base64.b64decode(data['data']).decode('utf-8')
+
+    # Load Data in JSON
+    json_string = json.loads(name)
```
Comment (ocervell) [inline examples/CloudFunction/gsuite-exporter-cloudfunction/main.py]: This is not a json_string but a dictionary --> rename variable

File: examples/CloudFunction/gsuite-exporter-cloudfunction/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+boto3
+gsuite_exporter==0.0.2
```
Comment (ocervell) [inline examples/CloudFunction/gsuite-exporter-cloudfunction/requirements.txt]: Increment version to latest version: 0.0.3.
Ideally, make this file generated by Terraform and add a `gsuite_exporter_version` variable to the Terraform defaulting to the latest version.

File: examples/CloudFunction/main.tf [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  name     = "${var.name}-gsuite-audit-log-scheduler"
+  description = "Scheduler for Gsuite Exporter Cloudfunction"
+  schedule = var.cs_schedule
+  time_zone = "America/Denver"
+
+  pubsub_target {
+    topic_name = "projects/${var.project_id}/topics/${google_pubsub_topic.trigger-topic.name}"
+    data = base64encode("{\"PROJECT_ID\":\"${var.project_id}\",\"GSUITE_ADMIN_USER\":\"${var.gsuite_admin_user}\"}")
+  }
+}
```
Comment (ocervell) [inline examples/CloudFunction/main.tf]: We have a module for Cloud Scheduler + Cloud Functions [here](https://github.com/terraform-google-modules/terraform-google-scheduled-function). Do you think you can leverage it instead ?

File: examples/CloudFunction/README.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
+# Terraform Stackdriver Aggregated Export to PubSub
```
Comment (ocervell) [inline examples/CloudFunction/README.md]: Please rename the example folder structure to lower-case style and better names:
`examples/CloudFunction/README.md` --> `examples/cloud_function/README.md`
`examples/CloudFunction/gsuite-exporter-cloud-function` --> `examples/cloud_function/code`
...

File: examples/CloudFunction/README.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+  * The following scopes in the [API client access page](https://admin.google.com/AdminHome?chromeless=1#OGX:ManageOauthClients)
+    - https://www.googleapis.com/auth/admin.reports.audit.readonly (to read from the Reports API)
+    - https://www.googleapis.com/auth/iam (to generate a super-admin token)
+
+## Usage
+The usage of the module within your own main.tf file is as follows:
+
+```hcl
+    module "aggregated-export-to-pubsub" {
```
Comment (ocervell) [inline examples/CloudFunction/README.md]: this won't do an aggregated export to pubsub.
Rename module name to `"gsuite-export-cloud-function"`

File: examples/CloudFunction/README.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+      source                          = ./path-to-your-source
```
Comment (ocervell) [inline examples/CloudFunction/README.md]: `./path-to-your-source` --> `terraform-google-modules/gsuite-export/google`

Add:
`version = "~> 1.0.0"`

File: examples/CloudFunction/gsuite-exporter-cloudfunction/README.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+The following destinations are currently supported:
+- [Stackdriver Logging](https://cloud.google.com/logging/docs/)
+
+## Requirements
+* A GSuite Admin account
+* A service account with:
+  * [GSuite domain-wide delegation](https://developers.google.com/admin-sdk/reports/v1/guides/delegation) enabled.
+  * The IAM role `roles/iam.tokenCreator` set on the organization.
+
+### Collectors
```
Comment (ocervell) [inline examples/CloudFunction/gsuite-exporter-cloudfunction/README.md]: Remove from here till the end - this is already present in the GSuite Exporter documentation.

**Review by ocervell — CHANGES_REQUESTED** [review]
Reviewed - good work so far ! Small changes to address. The module has no tests (will be added with #9), but if you want to give a shot at writing tests for your examples after #9 is merged, please do !

**Note for myself:** after merging this, I'll need to split this module into two submodules:
- `modules/compute_engine` --> the existing module
- `modules/cloud_function` --> this example (to turn into a module)

File: examples/CloudFunction/gsuite-exporter-cloudfunction/requirements.txt (hunk 3 above)
Comment (ocervell) [reply examples/CloudFunction/gsuite-exporter-cloudfunction/requirements.txt]:
```suggestion
gsuite_exporter==0.0.3
```

File: examples/CloudFunction/README.md (hunk 6 above)
Comment (ocervell) [reply examples/CloudFunction/README.md]:
```suggestion
    module "gsuite-export-cloud-function" {
```

File: examples/CloudFunction/README.md (hunk 7 above)
Comment (ocervell) [reply examples/CloudFunction/README.md]:
```suggestion
      source                          = "github.com/terraform-google-modules/terraform-google-gsuite-export/tree/master/examples/CloudFunction"
```

**Review by ocervell — COMMENTED** [review]
@cleibl Added some suggested commits

File: examples/CloudFunction/gsuite-exporter-cloudfunction/README.md (hunk 8 above)
Comment (cleibl) [reply examples/CloudFunction/gsuite-exporter-cloudfunction/README.md]: Removed

**Comment by cleibl** [comment]
@ocervell committed requested changes

**Pull request merged**